neo4j-ubuntu-vm/azuredeploy.json

```diff
@@ -0,0 +1,433 @@
+        "Premium_LRS"
+      ],
+      "metadata": {
+        "description": "Storage account type. Premium storage is recommended for graph databases. Make sure storage type is supporting by VM size."
+      }
+    },
+    "vmAdminUsername": {
+      "type": "string",
+      "defaultValue": "azureuser",
+      "minLength": 1,
+      "metadata": {
+        "description": "User name for the virtual machine."
+      }
+    },
+    "sshKeyData": {
+      "type": "string",
+      "metadata": {
+        "description": "SSH rsa public key file as a string."
+      }
+    },
+    "neo4jEdition": {
+      "type": "string",
+      "defaultValue": "neo4j",
+      "allowedValues": [
+        "neo4j",
+        "neo4j-enterprise",
+        "neo4j=3.2.1",
+        "neo4j-enterprise=3.2.1",
+        "neo4j=3.2.5",
+        "neo4j-enterprise=3.2.5",
+        "neo4j=3.3.0",
+        "neo4j-enterprise=3.3.0"
+      ],
+      "metadata": {
+        "description": "Neo4J edition and version to install."
+      }
+    },
+    "publicIPAddressDns": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Unique DNS name for public IP Address. As in: publicIPAddressDns.location.cloudapp.azure.com"
+      }
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/helshabini/neo4j-ubuntu-vm/master/"
+    },
+    "_artifactsLocationSasToken": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+      },
+      "defaultValue": ""
+    }
+  },
+  "variables": {
+    "vmImagePublisher": "Canonical",
+    "vmImageOffer": "UbuntuServer",
+    "storageaccountName": "[concat(parameters('storageaccountPrefix'), uniqueString(resourceGroup().id))]",
+    "storageaccountContainerName": "vhds",
+    "sshKeyPath": "[concat('/home/',parameters('vmAdminUsername'),'/.ssh/authorized_keys')]",
+    "scriptsFolder": "scripts/",
+    "virtualNetworkName": "[concat(parameters('vmName'), '-vnet')]",
+    "virtualNetworkPrefix": "10.0.0.0/16",
+    "virtualNetworkSubnetName": "default",
+    "virtualNetworkSubnetPrefix": "10.0.0.0/24",
+    "networkSecurityGroupName": "[concat(parameters('vmName'), '-nsg')]",
+    "networkInterfaceName": "[concat(parameters('vmName'), '-nic')]",
+    "networkInterfaceVnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+    "networkInterfaceSubnetRef": "[concat(variables('networkInterfaceVnetID'), '/subnets/', variables('virtualNetworkSubnetName'))]",
```

Comment:
simplify, just use resourceId()

Reply:
Could you elaborate or provide a different example?